python/ray/rllib/rollout.py

```diff
@@ -96,14 +97,24 @@ def run(args, parser):
     agent = cls(env=args.env, config=config)
     agent.restore(args.checkpoint)
     num_steps = int(args.steps)
-    rollout(agent, args.env, num_steps, args.out, args.no_render)
+    rollout(agent, args.env, num_steps, config, args.out, args.no_render)
 
 
-def rollout(agent, env_name, num_steps, out=None, no_render=True):
+def rollout(agent, env_name, num_steps, config, out=None, no_render=True):
     if hasattr(agent, "local_evaluator"):
         env = agent.local_evaluator.env
     else:
         env = gym.make(env_name)
+
+    if config['model']['use_lstm']:
+        use_lstm = True
+        state_init = [
```

Comment:
You can actually always set state_init to agent.local_evaluator.policy_map["default"].get_initial_state().

It will return empty list for non rnn policies.

Reply:
Ah, got it. I'll make this change; thanks!

Reply:
So, what if there isn't a local evaluator? Is there presumably always one?